test/kubetest/kubernetes.go

```diff
@@ -0,0 +1,332 @@
+package kubetest
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"gopkg.in/yaml.v2"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	kubeyaml "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes"
+)
+
+func CreatedManifests(client kubernetes.Interface, paths ...string) Setup {
+	return func(ctx *ScenarioContext) error {
+		for _, path := range paths {
+			content, err := ioutil.ReadFile(path)
+			if err != nil {
+				return err
+			}
+
+			if len(content) == 0 {
+				return fmt.Errorf("manifest has no content: %s", path)
+			}
+
+			var meta metav1.TypeMeta
+			if err = yaml.Unmarshal(content, &meta); err != nil {
+				return err
+			}
+
+			// TODO: This needs to be more generic!
+
+			kind := strings.ToLower(meta.Kind)
```

Comment:
Wasn't there a generic kube client that could apply arbitrary manifests?

Reply:
there's not, but I built a very very rudimentary one: https://github.com/brancz/uclient

Reply:
Earlier I asked @LiliC about such a tool and she recommended this one to me. Handy you already solved most of my problems @brancz :smile: 
I've already pushed that integrated to my fork. Works like a charm! :100: 

https://github.com/metalmatze/kube-rbac-proxy/blob/e021a74707f12859a671c400ffc1b73fa95f2a1b/test/kubetest/kubernetes.go#L43-L52